File: neetcode/Contains_Dup/containsDuplicate.js
let nums = [1,2,3,4,3];

function containsDuplicate(nums) {
    let numbers = new Set();
    for(let i = 0; i < nums.length; i++) {
        if(numbers.has(nums[i])) {
            return true
        }
        else {
            numbers.add(nums[i])
            console.log(numbers);
        }
    }
    return false;
};

console.log(containsDuplicate(nums));

File: neetcode/Contains_Dup/containsDuplicate.py
# //given an interger array nums, return true if any value appears at least twice in the array, and return false if there are no duplicates

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashset = set()

        for num in nums:
            if num in hashset:
                return True
            hashset.add(num)
        return False

